components/script/dom/htmlinputelement.rs

```diff
@@ -1071,7 +1081,23 @@ impl<'dom> LayoutHTMLInputElementHelpers<'dom> for LayoutDom<'dom, HTMLInputElem
 
         match self.input_type() {
             InputType::Checkbox | InputType::Radio => "".into(),
-            InputType::File | InputType::Image => "".into(),
+            InputType::File | InputType::Image => {
+                let filelist = self.get_filelist();
+                match filelist {
+                    Some(filelist) => {
+                        let length = filelist.Length();
+                        if length == 0 {
+                            return "No file chosen".into();
+                        }
+                        if length == 1 {
+                            return "1 file".into();
+                        }
+
+                        format!("{} files", length).into()
+                    },
+                    None => "No file chosen".into(),
+                }
+            },
```

Comment:
Not quite where I wanted it to be. Ideally I'd like to print the file name when there's only one file selected, but when I tried to access the file using `filelist.Item(0)` it'll simply panic.

If anyone knows how to get around this I'll add it in.

Reply:
One file should just display the file name.

Reply:
To avoid the panic you need to add layout helpers to FileList that can return a `&'dom File` from the internal list.

Reply:
> To avoid the panic you need to add layout helpers to FileList that can return a `&'dom File` from the internal list.

Thanks!